extensions/wikia/Chat2/Chat.class.php

```diff
@@ -72,435 +82,207 @@ public static function echoCookies() {
+			return wfMessage( 'chat-ban-you-need-permission', self::CHAT_MODERATOR )
 				->inContentLanguage()->text() . "\n";
 		}
 
 		// Make sure we aren't trying to kick/ban someone who shouldn't be kick/banned
-		if ( $userToKickBan->isAllowed( $kickBanPermission ) && !$kickingUser->isAllowed( 'chatstaff' ) && !$kickingUser->isAllowed( 'chatadmin' ) ) {
-			return wfMessage( 'chat-ban-cant-ban-moderator' )
-				->inContentLanguage()->text() . "\n";
-		}
-
-		self::banUserDB(
-			F::app()->wg->CityId,
-			$userToKickBan,
-			$kickingUser,
-			$time,
-			$reason,
-			$time == 0 ? 'remove': 'add'
-		);
-
-		return true;
-	}
-
-	/**
-	 * @param string $username
-	 * @param string $dir
-	 * @param User $kickingUser
-	 *
-	 * @return bool
-	 * @throws DBUnexpectedError
-	 */
-	public static function blockPrivate( $username, $dir = 'add', $kickingUser ) {
-		ChatHelper::info( __METHOD__ . ': Method called', [
-			'username' => $username,
-			'dir' => $dir,
-			'kickingUser' => $kickingUser,
-		] );
-
-		$kickingUserId = intval( $kickingUser->getId() );
-		$userToBlock = User::newFromName( $username );
-		$dbw = wfGetDB( DB_MASTER, [], F::app()->wg->ExternalDatawareDB );
-
-		if ( !empty( $userToBlock ) && $kickingUserId > 0 ) {
-			if ( !wfReadOnly() ) { // Change to wgReadOnlyDbMode if we implement thatwgReadOnly
-				if ( $dir == 'remove' ) {
-					$dbw->delete(
-						"chat_blocked_users",
-						[
-							'cbu_user_id' => $kickingUserId,
-							'cbu_blocked_user_id' => $userToBlock->getId()
-						],
-						__METHOD__
-					);
-				} else {
-					$dbw->insert(
-						"chat_blocked_users",
-						[
-							'cbu_user_id' => $kickingUserId,
-							'cbu_blocked_user_id' => $userToBlock->getId()
-						],
-						__METHOD__,
-						[ 'IGNORE' ]
-					);
-				}
-				$dbw->commit();
-			}
+		// Chat moderators can be kicked/banned only by staff members and admins
+		if ( !self::canBan( $subjectUser, $adminUser ) ) {
+			return wfMessage( 'chat-ban-cant-ban-moderator' )->inContentLanguage()->text() . "\n";
 		}
-		return true;
-	}
 
-	/**
-	 * Add, update or remove a user ban from chat on a specific wikia
-	 *
-	 * @param int $cityId The ID of the wikia where the user was banned
-	 * @param User $banUser The ID of the user who was banned
-	 * @param User $adminUser The ID of the user doing the banning if adding or updating
-	 * @param int $time The time in seconds to ban the user if adding or updating
-	 * @param string $reason Why the user ban status was changed
-	 * @param string $action The operation
-	 *
-	 * @throws DBUnexpectedError
-	 * @throws MWException
-	 */
-	public static function banUserDB( $cityId, $banUser, $adminUser, $time, $reason, $action = 'add' ) {
-		if ( empty( $banUser ) || empty( $adminUser ) ) {
-			return;
-		}
-
-		if ( wfReadOnly() ) {
-			return;
-		}
+		$cityId = F::app()->wg->CityId;
+		$action = $time != 0 ? self::ACTION_ADD : self::ACTION_REMOVE;
```

Comment:
why like that? could you please add comment above explaining why we want to remove users if time equals 0 and add if not?
also, could you please re-add params descriptions removed here? `
 -	 * @param int $cityId The ID of the wikia where the user was banned
 -	 * @param User $banUser The ID of the user who was banned
 -	 * @param User $adminUser The ID of the user doing the banning if adding or updating
 -	 * @param int $time The time in seconds to ban the user if adding or updating
 -	 * @param string $reason Why the user ban status was changed
 -	 * @param string $action The operation` where applicable?